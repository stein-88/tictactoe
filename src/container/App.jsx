import React from 'react'
import { BrowserRouter, Routes, Route, Navigate } from 'react-router-dom'
import TicTacToe from '../components/TicTacToe'
import Footer from '../components/Footer'

const App = () => {
    return (
        <BrowserRouter>
            <Routes>
                <Route element={<TicTacToe />} path="/" />
                <Route element={<TicTacToe handleContent={handleContent} />} path="/tictactoe" />
                <Route element={<Navigate replace to="/" />} path="*" />
            </Routes>
            <Footer />
        </BrowserRouter>
    )
}

export default App
