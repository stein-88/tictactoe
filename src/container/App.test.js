import React from 'react'
import renderer from 'react-test-renderer'
import App from './App'
import { Provider } from 'react-redux'
import configureStore from '../store'

jest.mock('uuid', () => ({ v4: () => `uniquekey${Math.random(1000) * 100}` }))

describe('Test on App container', () => {
	test('Renderized with no props', () => {
		const compNoProps = renderer.create(
			<Provider store={configureStore()}>
				<App />
			</Provider>
		)
		const compJson = compNoProps.toJSON()
		expect(compJson).toMatchSnapshot()
	})
})