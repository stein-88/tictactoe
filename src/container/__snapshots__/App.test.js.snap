// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on App container Renderized with no props 1`] = `
Array [
  <section
    className="w-100 bgNewlikeblue"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-lg-8 col-12 m-auto"
        >
          <div
            className="row"
          >
            <div
              className="col-lg-6 col-12"
            >
              <p
                className="h1 pb-5"
              >
                Tic Tac Toe
              </p>
              <p
                className="h3 py-3"
                data-test="players"
              >
                Player One: 
                <strong
                  className="text-danger"
                >
                  X
                </strong>
              </p>
              <p
                className="h3 "
                data-test="players"
              >
                Player Two: 
                <strong
                  className="text-danger"
                >
                  O
                </strong>
              </p>
            </div>
            <div
              className="col-lg-6 col-12"
            >
              <section
                className="w-100 text-center"
              >
                <section
                  className="w-100 px-5 mb-3"
                >
                  <img
                    alt="Tic Tac Toe"
                    className="w-100 img-fluid rounded"
                    src="test-file-stub"
                  />
                </section>
                <span
                  aria-hidden="true"
                  className="btnstyle"
                  onClick={[Function]}
                >
                  Start Game
                </span>
              </section>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer>
    <section
      className="py-5 bgblack"
      id="footer"
    >
      <p
        className="py-5 text-center text-white"
      >
        Â© 2022 Wanderson Santos. All Rights Reserved.
      </p>
    </section>
  </footer>,
]
`;
